### PR TITLE
Fix inconsistent cross origin error handling

### PR DESCRIPTION
**Pull request by luisrudge**
We weren't applying the error mapping that we do in other errors.

### CODE REVIEW

File: test/web-auth/cross-origin-authentication.test.js [hunk 1]
```diff
@@ -267,7 +267,11 @@ describe('auth0.WebAuth.crossOriginAuthentication', function() {
               anotherOption: 'foobar'
             },
             function(err) {
-              expect(err).to.be.eql({ error: 'any_error', error_description: 'any error' });
+              expect(err).to.be.eql({
+                original: { error: 'any_error', error_description: 'any error' },
+                code: 'any_error',
```
Comment (lbalmaceda) [inline test/web-auth/cross-origin-authentication.test.js]: I know that it's different `any_error` to `any error`, but can you please change one of them to something more visibly different? i.e. `code: 'any_error', description: 'a long error description'`

File: test/web-auth/cross-origin-authentication.test.js [hunk 2]
```diff
@@ -300,8 +304,12 @@ describe('auth0.WebAuth.crossOriginAuthentication', function() {
             },
             function(err) {
               expect(err).to.be.eql({
-                error: 'request_error',
-                error_description: '{"some":"error"}'
+                original: {
+                  error: 'request_error',
+                  error_description: '{"some":"error"}'
```
Comment (lbalmaceda) [inline test/web-auth/cross-origin-authentication.test.js]: are you sure this particular error description is an object?

File: src/web-auth/cross-origin-authentication.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ var windowHelper = require('../helper/window');
 var objectHelper = require('../helper/object');
 var RequestBuilder = require('../helper/request-builder');
 var WebMessageHandler = require('./web-message-handler');
+var responseHandler = require('../helper/response-handler');
```
[the remark this answers is not in the document]
Comment (luisrudge) [reply src/web-auth/cross-origin-authentication.js]: WebMessageHandler is a class, responseHandler is a function

File: test/web-auth/cross-origin-authentication.test.js (hunk 2 above)
Comment (luisrudge) [reply test/web-auth/cross-origin-authentication.test.js]: it's a string, not an object.

Comment (luisrudge) [reply test/web-auth/cross-origin-authentication.test.js]: https://github.com/auth0/auth0.js/blob/master/src/web-auth/cross-origin-authentication.js#L71

**Pull request merged**